### PR TITLE
Addressed issue #359

### PR DESCRIPTION
**Pull request by coverney**

### CODE REVIEW

File: code/TASBEConfig.m [hunk 1]
```diff
@@ -66,7 +66,9 @@
             doc.flow.conditionEventRatioWarning = 'Threshold to warn on variation in condition event counts.';
             s.flow.conditionEventRatioWarning = [];
             defaults('flow.conditionEventRatioWarning') = 'flow.eventRatioWarning';
-
+            doc.flow.conditionFracGatedWarning = 'Threshold to warn on variation in fraction of gated particles. Value should be in percent from 0 to 1.';
```
Comment (jakebeal) [inline code/TASBEConfig.m]: Remove "in percent", since it's fraction, not percentage.

File: code/per_color_constitutive_analysis.m [hunk 2]
```diff
@@ -153,3 +157,20 @@
...
+
+% warn when some samples have a fraction of gated particles that is very
+% different 
+median_fracs_removed = median(cell2mat(fracs_removed));
+lower_bound = (1-TASBEConfig.get('flow.conditionFracGatedWarning'))*median_fracs_removed;
+upper_bound = (1+TASBEConfig.get('flow.conditionFracGatedWarning'))*median_fracs_removed;
+for i=1:n_conditions
+   frac = results{i}.frac_removed;
+   if frac <= lower_bound
+       TASBESession.warn('TASBE:Analysis', 'HighGatingVariation', 'Fraction of gated particles in condition "%s" is much lower than the median', results{i}.condition);
```
Comment (jakebeal) [inline code/per_color_constitutive_analysis.m]: change "much lower" to "significantly lower", and add the fraction and the median to provide specific information. Do the same for the upper bound warning.

File: tests/test_batch_analysis.m [hunk 3]
```diff
@@ -220,8 +220,48 @@
...
+
+TASBEConfig.set('flow.conditionFracGatedWarning', 0.01);
+
+% Execute the actual analysis
+TASBEConfig.set('OutputSettings.StemName','LacI-CAGop');
+[results, sampleresults] = per_color_constitutive_analysis(CM,file_pairs,{'EBFP2','EYFP','mKate'},AP);
+
+% check for warning
+log = TASBESession.list();
+assertEqual(log{end}.contents{end}.name, 'HighGatingVariation');
```
Comment (jakebeal) [inline tests/test_batch_analysis.m]: Check that you get the right number of warnings and that they've got the right values.

File: code/TASBEConfig.m (hunk 1 above)
Comment (coverney) [reply code/TASBEConfig.m]: Done

File: code/per_color_constitutive_analysis.m (hunk 2 above)
Comment (coverney) [reply code/per_color_constitutive_analysis.m]: Done

File: tests/test_batch_analysis.m (hunk 3 above)
Comment (coverney) [reply tests/test_batch_analysis.m]: By right values, do you mean the fractions and median? If so, then I am thinking of parsing through the warning message to get those values.

Comment (coverney) [reply tests/test_batch_analysis.m]: Done

**Pull request merged**